framework/src/base/NonlinearSystem.C

```diff
@@ -325,39 +322,38 @@ NonlinearSystem::stopSolve()
 void
 NonlinearSystem::initialSetup()
 {
-  for (unsigned int i=0; i<libMesh::n_threads(); i++)
-    _constraints[i].initialSetup();
-}
-
-void
-NonlinearSystem::initialSetupBCs()
-{
-  for (unsigned int i=0; i<libMesh::n_threads(); i++)
-    _bcs[i].initialSetup();
-}
-
-void
-NonlinearSystem::initialSetupKernels()
-{
-  for (unsigned int i=0; i<libMesh::n_threads(); i++)
+  for (THREAD_ID tid = 0; tid < libMesh::n_threads(); tid++)
   {
-    _kernels[i].initialSetup();
-    _dirac_kernels[i].initialSetup();
-    if (_doing_dg) _dg_kernels[i].initialSetup();
+    _kernels.initialSetup(tid);
+    _dirac_kernels.initialSetup(tid);
+    if (_doing_dg)
+      _dg_kernels.initialSetup(tid);
+
+    _dampers.initialSetup(tid);
+    _integrated_bcs.initialSetup(tid);
   }
+  _scalar_kernels.initialSetup();
+  _constraints.initialSetup();
```

Comment:
Constraints are not threaded and it is not resized.

https://github.com/idaholab/moose/blob/devel/framework/src/base/NonlinearSystem.C#L879
https://github.com/idaholab/moose/blob/devel/framework/src/base/NonlinearSystem.C#L902